From 825102a36202c4d94009a997ad4c3bf8420a4e88 Mon Sep 17 00:00:00 2001
From: shoya0x00 <shoya0x00@gmail.com>
Date: Sat, 19 Oct 2024 12:42:04 +0000
Subject: [PATCH 2/2] Settings: Set correct category and styling for google and
 digital wellbeing

On new revamped Settings UX dashboard missing styling for google and digital wellbeing tile.
This fixes the styling and move these apps tiles to their respective category

Change-Id: I5f18fab2419c8346760c4f6bed6f77e1eac8e059
Signed-off-by: shoya0x00 <shoya0x00@users.noreply.github.com>
---
 .../settings/dashboard/DashboardFragment.java       | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/com/android/settings/dashboard/DashboardFragment.java b/src/com/android/settings/dashboard/DashboardFragment.java
index 9abc6c2..dca60f9 100644
--- a/src/com/android/settings/dashboard/DashboardFragment.java
+++ b/src/com/android/settings/dashboard/DashboardFragment.java
@@ -556,12 +556,17 @@ public abstract class DashboardFragment extends SettingsPreferenceFragment
                         screen.addPreference(pref);
                     }
                 } else {
+                    Preference group = null;
                     if (tile.hasGroupKey()
                             && mDashboardTilePrefKeys.containsKey(tile.getGroupKey())) {
-                        Preference group = screen.findPreference(tile.getGroupKey());
-                        if (group instanceof PreferenceCategory) {
-                            ((PreferenceCategory) group).addPreference(pref);
-                        }
+                        group = screen.findPreference(tile.getGroupKey());
+                    } else if ("top_level_google".equals(key)) {
+                        group = screen.findPreference("top_level_account_category");
+                    } else if ("top_level_wellbeing".equals(key)) {
+                        group = screen.findPreference("top_level_security_privacy_category");
+                    }
+                    if (group instanceof PreferenceCategory) {
+                        ((PreferenceCategory) group).addPreference(pref);
                     } else {
                         screen.addPreference(pref);
                     }
-- 
2.34.1
